### DISCUSSION TITLE
Toggle dark mode without being logged in

### DISCUSSION THREAD

**Discussion by fyliu** [community/community#33190] [Other Feature Feedback, Questions, & Ideas] [answered]
It would be great (if not already available) to be able to toggle dark mode without having to log in.

Use case: I default to opening links in anonymous browsers that I prefer to not have logged in to anything. My system theme is dark all the time, but I would like to view websites in light mode during the day or if I have lights on. Manual toggling is the best option for me.

Options is important to users like me.

Having Github always display in light mode is better in a way, since users would have the option of using browser extensions to enable dark mode. I don't believe there are browser extensions to "turn on the light".

**Comment by benjiallen** [comment]
@fyliu thanks for the feedback.

I don't think you can manually toggle themes when you are logged out. As you have observed, GitHub will detect operating system settings and try to guess a good theme when you are logged out.

**Reply by tilusnet** [reply]
An optimal theme should be subject to one's indoor light conditions, and not the time of the day!
Please disable this "smart" feature, it's really annoying!

**Reply by adamency** [reply]
This is absolutely a complete anti-feature. Stop guessing what users want, and even worse, forcing your guess on them. Let them choose themselves.

**Reply by adamency** [reply]
Have some dignity and stand by (the very first line of) [your own profile description](https://github.com/benjiallen)...

> I care about product management, digital accessibility, and good user experience.

**Reply by delanym** [reply]
Its a reasonable explanation though. As long as you're not logged in you're a bot not a user

**Reply by adlai** [reply]
> [...] you're not logged in you're a bot not a user

have you ever copied any keys passwords or safe application slogons?

forex ample, my favorite professor almost epileptically pissed me pants to the point by brushing aside the reason for our conversation with the words, "use the app!"!

I could have spat coffee.

**Comment by peter31990** [comment]
How can you turn off the ugly dark mode without logging in? Dark theme is unpleasant to the eyes and should not be used by default and without the option to turn it off.

**Reply by adlai** [reply]
your browser might be forcing it using third-party scripts; try disabling any you find, especially if you installed ANY scheduling assistants.

**Comment by tilusnet** [comment]
Please allow not logged in users to toggle dark/light mode.

**Comment by james-bern** [comment]
I would also be very excited to have this feature. I use Github to post course notes, which are harder for many students to read in dark mode.

**Reply by cynber** [reply]
That's the issue with me as well. Some `.png` images are illegible with the default dark theme. I'd imagine there are other accessibility issues as well with course notes.

A simple toggle in the UI would help a lot.

**Comment by jonny-379** [comment]
Yes please let us disable the dark mode without login! 
I have serious trouble reading after browser github for "long time" (more then 10 minutes). You should not be required to sig up or in to change it.

**Reply by adlai** [reply]
too many people on the social networking side of the merkle chain starter train

**Comment by tomoveu** [comment]
Forcing dark mode on users is not nice, nor useful.

**Reply by adlai** [reply]
sad up vote

**Comment by james-bern** [comment]
Hi. Checking back in on this. I use Github to host notes and documentation for my courses, which are harder for students to read in dark mode.

**Reply by adlai** [reply]
have you studied acksessibility of screenshots ? document formats might include the image with alpha channel variably defined in terms of palette, rather than hardcoded like daguerrotypes and polaroids.

**Comment by matheusqmello** [comment]
If it makes sense for you, you can edit 1 attribute of "html" tag on github website, using the browser developer tool/inspect sourcecode:
- `data-color-mode="light"` --> change to `"dark"` to see git in dark theme and vice-versa;

Example:
`<html lang="en" data-color-mode="light" data-light-theme="light" data-dark-theme="dark" data-a11y-animated-images="system" data-a11y-link-underlines="true" data-turbo-loaded="">`

**Reply by imme-emosol** [reply]
bookmarklet:

    javascript:(()=>{document.querySelectorAll('html').dataset.colorMode='dark';})();

**Comment by CodeByMrFinchum** [comment, marked as answer]
I made a Chrome plugin aviable on my [gitlab](https://gitlab.com/CodeByMrFinchum/github-light-mode-enforcer) or directly here on [github](https://github.com/CodeByMrFinchum/github-light-mode-enforcer) (simply a copy), that changes the "auto" to "light" thereby making github light even if the brower / OS theme is set to dark, without needing an account.

It is not on the chrome store, so you have to install it manually and set developer mode in chrome (Brave, Vivaldi, etc), I used GNU Affero General Public License v3.0 so you can just fork and publish it if you want to, you only need to keep GNU v3.0 as license.

Lets see if Microsoft is gonna dislike this and remove it.

**Reply by fyliu** [reply]
I'm going to mark this as the answer since it feels like this is as good as we'll get. I wonder if I can also mark the [bookmarklet comment](https://github.com/orgs/community/discussions/33190#discussioncomment-12786040) as another answer. I guess not. Well, they both look good to me.
